### PR TITLE
BaSP-MR-145: Implement react-hook-form with joi validations in Classes

### PR DESCRIPTION
**Pull request by fernandofernandezuy**
This PR introduces the integration of the react-hook-form library in the entity of classes, along with the joi schema validation and the @hookform/resolvers/joi resolver, to handle the form more efficiently.

With react-hook-form, the form now displays validation errors when you move away from each input. This provides immediate feedback on any incorrect or missing data.

Also I corrected styles in the shared input component, and added a prop to specify the type in the shared button.

### CODE REVIEW

**Comment by fernandofernandezuy** [comment]
[[Classes] - React-hook-form implementation](https://trello.com/c/T9wifyiI/145-classes-react-hook-form-implementation)

File: src/Components/Classes/Form/index.jsx [hunk 1]
```diff
@@ -56,38 +44,56 @@ const Form = () => {
...
-  const createClass = async () => {
+  const createClass = async (data) => {
     try {
-      const data = await dispatch(addClass(gymClass));
-      showSuccesModal(data);
+      const response = await dispatch(addClass(data));
+      showSuccesModal(response);
     } catch (error) {
       showErrorModal(error);
+      console.log(data);
```
Comment (EmaBaldor) [inline src/Components/Classes/Form/index.jsx]: This shouldn't go here

**Review by EmaBaldor — COMMENTED** [review]
Great job!

**Review by EmaBaldor — APPROVED** [review]

**Review by Gonzarey96 — APPROVED** [review]
Great job

File: src/Components/Classes/Form/form.module.css [hunk 2]
```diff
@@ -46,10 +46,15 @@
 
 .buttonsDiv {
   display: flex;
+  flex-direction: column;
   justify-content: center;
   align-items: center;
 }
 
+
```
Comment (ghost) [inline src/Components/Classes/Form/form.module.css:54]: great job Fer!!!, if you have to make a new commit for another correction take the opportunity to delete these blank lines.

**Review by ghost — APPROVED** [review]

**Comment by araefedg** [comment]
Hi Fer! i found some bugs testing your implementations, left a ticket on Trello: https://trello.com/c/6tdxuBN1

**Pull request merged**